### ISSUE TITLE
[Question] joint_tests vs. anova/car::Anova over the GLM (for GEE they agree)

### ISSUE THREAD

**Issue by hugesingleton** [rvlenth/emmeans#339]
Hi! I'm posting it not knowing, if it's an issue, so titled in question rather.

I have simple dataset symbolising a longitudinal study with 4 time points and successes of some procedure. I want to model the proportions over time using the fact they there are repeated observations. In addition, I want to assess the main effect of time, via analysis of deviance.

As long as I use the geepack::geeglm - joint_tests and anova agree, approximately, but very closely, so it's OK. When I switch to GLM (inappropriately, I know, it ignores the clustering, but just wanted to compare the outputs), they differ. And there's nothing fancy in this example, just single categorical IV.

Let's see the examples.

The data:
```{r}
d <- structure(list(ID = c(1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 
4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8, 9, 9, 
9, 9, 10, 10, 10, 10), TimeOrd = structure(c(1L, 2L, 3L, 4L, 
1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 
1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 
1L, 2L, 3L, 4L), .Label = c("M0", "M1", "M2", "M3"), class = c("ordered", 
"factor")), Result = c(1, 1.1, 1.5, 2, 1.1, 1.1, 1.4, 3, 1.1, 
1.2, 1.7, 5, 1.5, 3, 4.6, 6, 2.3, 4.4, 5.5, 8, 2, 2.2, 2.3, 7, 
4.5, 1, 3, 7, 3.3, 4.4, 5.5, 5, 3, 4, 5, 8, 1.1, 2.2, 3.3, 5), 
    SucN = c(0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 
    0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 
    1, 1, 1, 1, 0), Time = structure(c(1L, 2L, 3L, 4L, 1L, 2L, 
    3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 
    2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 1L, 2L, 3L, 4L, 
    1L, 2L, 3L, 4L), .Label = c("M0", "M1", "M2", "M3"), class = "factor")), row.names = c(NA, 
40L), class = "data.frame")
```

GEE:
```{r}
library(geepack)
library(emmeans)

m <- geeglm(SucN ~ Time, family=binomial(link = "logit"), data=d, id=ID, corstr = "unstructured", std.err="san.se")

> emmeans(m, specs=~Time, type="response")
 Time prob     SE  df asymp.LCL asymp.UCL
 M0    0.1 0.0949 Inf     0.014     0.467
 M1    0.3 0.1449 Inf     0.100     0.624
 M2    0.6 0.1549 Inf     0.297     0.842
 M3    0.7 0.1449 Inf     0.376     0.900

Covariance estimate used: vbeta 
Confidence level used: 0.95 
Intervals are back-transformed from the logit scale 
> 
> joint_tests(m)
 model term df1 df2 F.ratio p.value
 Time         3 Inf    2.18 0.0886 

> anova(m)
Analysis of 'Wald statistic' Table
Model: binomial, link: logit
Response: SucN
Terms added sequentially (first to last)

     Df   X2 P(>|Chi|)  
Time  3 6.53     0.089 
```
p-values agree closely.

Now let's run the GLM:

```{r}
m <- glm(SucN ~ Time, family=binomial(link = "logit"), data=d)

> anova(m, test = "Chisq")
Analysis of Deviance Table
Model: binomial, link: logit
Response: SucN
Terms added sequentially (first to last)

     Df Deviance Resid. Df Resid. Dev Pr(>Chi)  
NULL                    39       54.5           
Time  3     10.2        36       44.4    0.017 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

> car::Anova(m)
Analysis of Deviance Table (Type II tests)
Response: SucN
     LR Chisq Df Pr(>Chisq)  
Time     10.2  3      0.017 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

> joint_tests(m)
 model term df1 df2 F.ratio p.value
 Time         3 Inf    2.54 0.0545 
```

Here the p-values differ quite visible. At the same time, it's written in the manual, that joint_tests does what does SAS with type=3 analysis of deviance. And for a single covariate and balanced design, the type1=type2=type3.

So, is this a bug, or does the joint_tests do something more? The predictions are just the probabilities.

**Comment by rvlenth** [comment]
Your anova for the `glm` model is a likelihood-ratio test based on an asymptotic chi-square approximation, whereas the one in `joint_tests` is a Wald test based on an F approximation. That makes a difference. 

Another important fact is that anova results are comparisons of *models*, while `joint_tests` is based on one and only one model. To see exactly what is tested, try this, proceeding in small steps:
```r
RG = ref_grid(m)    # for emphasis -- everything emmeans does is based on the ref. grid
EMM = emmeans(RG, "Time")
CON = contrast(EMM, "consec")
```
If you do
```r
test(CON)
```
you see tests of the individual contrasts; whereas if you do
```r
test(CON, joint = TRUE)
```
you see a joint test of all of those contrasts jointly, and it is the same as what you get from `joint_tests()`. Both of the above tests are based on the linear functions of regression coefficients *from this one model* that are associated with these contrasts. You can see those linear functions via
```r
CON@linfct
```
The joint test of `CON` is based on constructing a quadratic form involving the above matrix and the inverse of `CON@V`, the estimated variance-covariance matrix of the regression coefficients. The individual tests of each contrast are equivalent to doing the same, but taking only one row of the matrix at a time. Again, this is quite different from what `anova(m)` does, which is to compare the deviances of the model `m` and another model having just the intercept.

The bottom line: They are different tests; thus the results can't be expected to be the same.

BTW, regarding type1 = type2 = type3 for a balanced design, that is only true with respect to *linear regression* models with *homogeneous errors*. And it is immaterial anyway because that refers primarily to *model-comparison* tests of multi-factor designs.

*Suggestion:* when reporting issues, I'd appreciate your not re-using the same symbol for different objects. That makes it confusing, for example, to talk about results of `anova(m)` -- which `m` are we talking about?

**Comment by hugesingleton** [comment]
Thank you very much for the explanation!
